### DISCUSSION TITLE
Going to start working on these Embeds, let us know your thoughts

### DISCUSSION THREAD

**Discussion by zeekhuge** [outline/outline#5938] [Ideas]
Hi @tommoor 

I & @PrikshitSingh24 are working on the following embeds
1. LinkedIn (already submitted at #5933) 
2. Pinterest
3. Github Projects
4. Github Repo
5. Windmill Apps (Self hosted + https://www.windmill.dev/)
6. Nocodb (Self hosted)
7. Zoho Calendar
8. Cal.com (self hosted + https://app.cal.com)

Please let us know if you seen any obvious problems here or/if we should drop some/all of these in case they don't align with the project goal, etc.

**Comment by tommoor** [comment]
Main thing to recognize is that PR's will only be accepted when:

- The service supports a proper embedded view (I don't think this is the case for eg GitHub Projects?)
- A regular link can be converted to an embedded link through string manipulation

**Reply by zeekhuge** [reply]
OK. Got it.

**Reply by alkanna** [reply]
So there are no plans to create custom cards for now for websites that would only support pulling information through their API ?

**Reply by Limezy** [reply]
I guess that this would mean a heavier codebase, both in terms of compiling / running ressources and cost / time for maintenance...

**Comment by prikshitsingh24** [comment]
@tommoor 
This is how windmill apps embed looks like, should I continue and add the embed of self hosted windmill as well?
![Screenshot from 2023-10-19 20-25-31](https://github.com/outline/outline/assets/99068054/5600114e-1620-4980-b02c-1fd637767a28)

**Reply by tommoor** [reply]
It's literally never been requested, I would wait until there is actually demand for these as all of these add extra code and maintenance. At some point in the next 6mo you will be able to embed anything without needing a specific integration for each, just waiting on #3000

**Reply by zeekhuge** [reply]
Well, in that case @PrikshitSingh24 let's just create the PR of the embeds we have already completed. That is:
1. Pinterest
2. Instagram

And let's ignore Windmill.

Does that sound OK @tommoor  ?

**Comment by SaadBazaz** [comment]
What's the stat on NocoDB? Would love to see this in action. 

Related: https://github.com/outline/outline/discussions/6507

**Reply by zeekhuge** [reply]
Well, @tommoor was working on JSON based rendering (instead of markdown) and that would allow any embed. This JSON based feature was supposed to come out in about 6 months from that last discussion, so we ditched working on any further embeds.

**Reply by SaadBazaz** [reply]
@zeekhuge - I don't see much progress so far (as per the Issue!) for JSON-based rendering. Until then, Nocodb is probably the biggest one in the list which makes sense. I could help build this if you have clues.

**Reply by zeekhuge** [reply]
@SaadBazaz Adding a new embed is simple. You can check the PRs added for other embeds by @PrikshitSingh24, for example.

However, the imp point here is - will @tommoor accept those PRs ?

**Reply by SaadBazaz** [reply]
Good point @zeekhuge.
will u, @tommoor? ;)

**Reply by tommoor** [reply]
While the JSON rendering is still coming would accept a NocoDB integration PR

**Reply by SaadBazaz** [reply]
I think I see JSON Rendering in the latest self-hosted version. 

If true, now's the time to make these @zeekhuge .
